labelbox/schema/confidence_presence_checker.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, List, Union
+
+from labelbox.data.annotation_types.annotation import (
+    ClassificationAnnotation, ObjectAnnotation, VideoClassificationAnnotation,
+    VideoObjectAnnotation)
+from labelbox.data.annotation_types.classification.classification import (
+    Checklist, ClassificationAnswer, Dropdown, Radio, Text)
+from labelbox.data.annotation_types.label import Label
+from labelbox.data.annotation_types.metrics.confusion_matrix import \
+    ConfusionMatrixMetric
+from labelbox.data.annotation_types.metrics.scalar import ScalarMetric
+from labelbox.data.serialization.ndjson.converter import NDJsonConverter
```

Comment:
There are two main parts of the sdk. labelbox/data and labelbox/schema. data brings with it a whole bunch of dependencies that not all users might want (e.g. if they just want to make simple api calls without all of the utilities they would only install labelbox/schema). You can't require these modules in labelbox/schema or it breaks the pattern. 

Another consideration is that Pydantic is also not the fastest lib in the world so we don't want to keep converting between versions if it isn't necessary. 

Lastly, I think we want to minimize the number of client side checks. It allows for an inconstant experience (e.g. create_from_url wouldn't even benefit from this) and also it is easy to go stale / has to be maintained.

imo we shouldn't apply this check.

Reply:
If you really want a check, you could do something as simple as a regex on the json string for a confidence key anywhere and that would be much more performant and simple to maintain.